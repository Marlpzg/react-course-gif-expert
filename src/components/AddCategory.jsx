import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length > 0) {
            onAddCategory(inputValue.trim())
            setInputValue('')
        }
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input type="text" value={inputValue} placeholder="Buscar GIFs" onChange={handleChange}/>
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
}