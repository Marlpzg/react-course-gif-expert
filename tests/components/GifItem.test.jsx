import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Pruebas sobre <GifItem />', () => {

    const title = 'Test Title';
    const url = 'https://media1.giphy.com/media/lfpwRGOHLZXsA/giphy.gif?cid=d21c202cga15slgpuzmt1ghp2e5e20pkqhdearnm7i67etid&ep=v1_gifs_search&rid=giphy.gif&ct=g';


    test('Debe coincidir con el Snapshot', () => {
        const { container } = render(<GifItem
            title={title}
            url={url}
        />);
        expect(container).toMatchSnapshot();
    })

    test('Debe coincidir mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem
            title={title}
            url={url}
        />);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title)
        //expect(container).toMatchSnapshot();
    })

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem
            title={title}
            url={url}
        />);
        expect(screen.getByText(title)).toBeTruthy();
        //expect(container).toMatchSnapshot();
    })
})