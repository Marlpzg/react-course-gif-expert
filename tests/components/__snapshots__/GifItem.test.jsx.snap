// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifItem /> Debe coincidir con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="https://media1.giphy.com/media/lfpwRGOHLZXsA/giphy.gif?cid=d21c202cga15slgpuzmt1ghp2e5e20pkqhdearnm7i67etid&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
